code/HUXt.py

```diff
@@ -441,6 +464,101 @@ def solve(self, cme_list=[], save=False, tag=''):
             self.save(tag=tag)
         return
     
+    @jit(nopython=False)
+    def generate_v_boundary_ts(self, cme_list=[]):
+        """
+        Genreate the input time series for the given longitudinla solar wind and list of cone cmes.
+        :param cme_list: A list of ConeCME instances to insert into the model.
+        """
+        
+        # Check only cone cmes in cme list
+        cme_list_checked = []
+        for cme in cme_list:
+            if isinstance(cme, ConeCME):
+                cme_list_checked.append(cme)
+            else:
+                print("Warning: cme_list contained objects other than ConeCME instances. These will be excluded")
+        
+        self.cmes = cme_list_checked
+                
+        constants= huxt_constants()
+        buffertime=constants['buffertime'].to(u.s)
+        self.model_time = np.arange(-buffertime.value, (self.simtime.to('s') + self.dt).value, self.dt.value) * self.dt.unit
+
+        dlondt = self.twopi * self.dt / self.synodic_period
+        lon, dlon, Nlon = longitude_grid()
+
+        # How many radians of Carrington rotation in this simulation length
+        simlon = self.twopi * self.simtime / self.synodic_period
+        # How many radians of Carrington rotation in the spin up period
+        bufferlon = self.twopi * buffertime / self.synodic_period
+        # Find the Carrigton longitude range spanned by the spin up and simulation period,
+        # centered on simulation longitude
+        lonint = np.arange(self.lon.value-bufferlon, self.lon.value + simlon+dlondt, dlondt)
+        # Rectify so that it is between 0 - 2pi
+        loninit = _zerototwopi_(lonint)
+        # Interpolate the inner boundary speed to this higher resolution
+        vinit = np.interp(loninit, lon.value, self.v_boundary.value, period=self.twopi) * self.kms
+        # convert from cr longitude to time
+        self.vinput_ts_amb = np.flipud(vinit).copy()
+        self.vinput_ts_cme = np.flipud(vinit).copy()
+
+        # Add the CME(s)
+        # ----------------------------------------------------------------------------------------
+        for t, time in enumerate(self.model_time):            
+            # Compute boundary speed of each CME at this time. Set boundary to the maximum CME speed at this time.
+            if time > 0:
+                if len(cme_list_checked) != 0:
+                    v_update_cme = np.zeros(len(cme_list_checked)) * self.kms
+                    for i, c in enumerate(cme_list_checked):
+                        r_boundary = self.r.min().to(u.km)
+                        v_update_cme[i] = _cone_cme_boundary_1d_(r_boundary, self.lon, time, self.vinput_ts_cme[t], c)
+                    self.vinput_ts_cme[t] = v_update_cme.max()
+    
+    @jit(nopython=False)
+    def solve_opt(self, cme_list, save=False, tag=''):
```

Comment:
I wonder if we should add a test function to check the output of HUXt1D.solve() and HUXt1D.solve_opt() are close enough to each other? It might be useful in the future if we make further changes and want be sure things are still consistent.